internal/dbtest/db.go

```diff
@@ -40,6 +41,38 @@ func Close() error {
 	return db.Close()
 }
 
+// InitWithMetadata initializes a database with optional metadata and DB file creation.
+// If meta is nil, no metadata file is created.
+// If createDBFile is false, no DB file is created (useful for testing "DB not found" scenarios).
+// Returns the dbDir path.
+func InitWithMetadata(t *testing.T, meta *metadata.Metadata, createDBFile bool) string {
+	t.Helper()
+
+	cacheDir := t.TempDir()
+	dbDir := db.Dir(cacheDir)
+
+	// Create metadata if provided
+	if meta != nil {
```

Comment:
nit: I don’t like using pointers when they aren’t needed.
Metadata is comparable, so we can use lo.IsEmpty:
https://go.dev/play/p/muF4oIjA9js?v=

But up to you.